RIOT-OS/RIOT
# cpu/esp*: esp_now and esp_wifi cleanup

### Contribution description

This PR is a cleanup of modules `esp_now` and `esp_wifi` to get them working together, for example to realize a border router which connects an ESP-NOW based mesh network of ESP32 an ESP8266 nodes with an infrastructure mode WiFi network.

### Testing procedure

Build a border router using project [RIOT_ESP_NOW_WiFi_Border_Router](https://github.com/gschorcht/RIOT_ESP_NOW_WiFi_Border_Router) and some ESP-NOW nodes. Follow the instructions in [README.md](https://github.com/gschorcht/RIOT_ESP_NOW_WiFi_Border_Router).

Although the test could be performed on either ESP8266 nodes or ESP32 nodes only, a configuration is recommended in an ESP32 is used as border router and ESP8266s are used as ESP NOW nodes.
[Update] ~However, if an ESP8266 is used as a boundary router, RPL must not be enabled due to its limited RAM.~ After merging PR # 10981, the limited amount of RAM for ESP8266 should no longer be a problem.

### Issues/PRs references

~Since it cleans up `esp_wifi` for ESP32 as well as ESP8266, it depends on PR #9917.~
